lib/internal/streams/end-of-stream.js

```diff
@@ -42,7 +42,8 @@ function eos(stream, opts, callback) {
     if (!readable) callback.call(stream);
   };
 
-  var readableEnded = stream._readableState && stream._readableState.endEmitted;
+  var readableEnded = stream.readableEnded ||
```

Comment:
I don’t think this change has a test. Would you mind adding one?

Reply:
As far as I can see there are no tests for end-of-stream whatsoever?

Reply:
> As far as I can see there are no tests for end-of-stream whatsoever?

 
test-stream-finished, test-stream-pipeline, and several others seem to cover the file incidentally?

Reply:
`test-stream-finished` are the tests for eos.